vms/platformvm/block/executor/block.go

```diff
@@ -29,25 +29,49 @@ func (*Block) ShouldVerifyWithContext(context.Context) (bool, error) {
 	return true, nil
 }
 
-func (b *Block) VerifyWithContext(_ context.Context, ctx *smblock.Context) error {
-	pChainHeight := uint64(0)
-	if ctx != nil {
-		pChainHeight = ctx.PChainHeight
+func (b *Block) VerifyWithContext(ctx context.Context, blockContext *smblock.Context) error {
+	var pChainHeight uint64
+	if blockContext != nil {
+		pChainHeight = blockContext.PChainHeight
 	}
 
 	blkID := b.ID()
 	if blkState, ok := b.manager.blkIDToState[blkID]; ok {
 		if !blkState.verifiedHeights.Contains(pChainHeight) {
-			// PlatformVM blocks are currently valid regardless of the ProposerVM's
-			// PChainHeight. If this changes, those validity checks should be done prior
-			// to adding [pChainHeight] to [verifiedHeights].
+			// Only the validity of warp messages need to be verified because
+			// this block has already passed verification with a different
+			// height.
```

Comment:
I tried to make the comment a bit more clear with:
```golang
// Only the validity of warp messages need to be verified because
// this block was already executed with a different block context.
```